Packs/RadwareCloudDDoSProtectionServices/Integrations/RadwareCloudDDoSProtectionServices/RadwareCloudDDoSProtectionServices.py

```diff
@@ -0,0 +1,237 @@
+            end_time(int): The end time for fetching alerts in milliseconds since epoch.
+            skip(int): The number of alerts to skip for pagination (default: 0).
+            take(int): The number of alerts to take for pagination (default: 700).
+        return: A dictionary containing the fetched alerts and related metadata.
+        """
+
+        params = {
+            "criteria": [
+                {
+                    "key": "timestamp",
+                    "value": [start_time, end_time]
+                },
+                {
+                    "key": "severity",
+                    "value": ["INFO", "LOW", "MEDIUM", "HIGH", "CRITICAL"]
+                }
+            ],
+            "skip": skip,
+            "take": take
+        }
+
+        return self._http_request(
+            method="POST",
+            url_suffix="/api/sdcc/infrastructure/core/analytics/object/operationalmessages/virtual",
+            json_data=params,
+        )
+
+
+def add_fields(events: list[dict], evnet_type: str | None):
+    """
+    Gets a list of events of a specific event type and adds the `_time` & `source_log_type` fields to the event.
+    Args:
+        events: A list of events.
+        evnet_type: The event type.
+    """
+    for event in events:
+        event['_time'] = event.get('timestamp')
+        event['source_log_type'] = evnet_type
+
+
+def fetch_events(client, last_run):
+    end_time = int(time.time() * 1000)
+    start_time = last_run.get('last_fetch_events', end_time-2)+1
+    continue_fetch_events = last_run.get('continue_fetch_events', None)
+    skip = 0
+    if continue_fetch_events:
+        end_time = continue_fetch_events.get('end_time')
+        start_time = continue_fetch_events.get('start_time')
+        skip = continue_fetch_events.get('fetched_events')
+
+    demisto.debug(f'RadwareCloudDDoS: {start_time=}, {end_time=}, {skip=}')
+    response = client.get_events(start_time, end_time, skip, 700)
+    documents = response.get("documents")
+    demisto.debug(f'RadwareCloudDDoS: {len(documents)=}')
+    new_continue_fetch_events = {}
+
+    if documents:
+        latest_timestamp_events = documents[0]["endTimestamp"]
+        demisto.debug(f'RadwareCloudDDoS: {latest_timestamp_events=}')
+        if len(documents) == 700:
+            demisto.debug('RadwareCloudDDoS: found next page')
+            new_continue_fetch_events = {'end_time': end_time, 'start_time': start_time, 'fetched_events': len(documents)+skip}
+
+        if not continue_fetch_events:
+            last_run['last_fetch_events'] = latest_timestamp_events
+            demisto.debug(f'RadwareCloudDDoS: saved {latest_timestamp_events=}')
+
+        if new_continue_fetch_events:
+            last_run['nextTrigger'] = '0'
+
+        last_run['continue_fetch_events'] = new_continue_fetch_events
+        demisto.debug(f'RadwareCloudDDoS: set {new_continue_fetch_events=}')
```

Comment:
Why not move this inside the condition?
```suggestion
        if new_continue_fetch_events:
            last_run['nextTrigger'] = '0'
            last_run['continue_fetch_events'] = new_continue_fetch_events
            demisto.debug(f'RadwareCloudDDoS: set {new_continue_fetch_events=}')
```

Reply:
It's there because it's also what resets it when it's not needed.